herald/herald/Sensor/BLE/BLEReceiver.swift

```diff
@@ -248,10 +248,14 @@ class ConcreteBLEReceiver: NSObject, BLEReceiver, BLEDatabaseDelegate, CBCentral
      */
     private func taskScanForPeripherals() {
         // Scan for peripherals -> didDiscover
-        var scanForServices : [CBUUID] = [BLESensorConfiguration.serviceUUID]
-        // Optionally include legacy advert only protocol as scan criteria
-        if let legacyAdvertOnlyProtocolServiceUUID = BLESensorConfiguration.legacyAdvertOnlyProtocolServiceUUID {
-            scanForServices.append(legacyAdvertOnlyProtocolServiceUUID)
+        var scanForServices: [CBUUID] = [BLESensorConfiguration.serviceUUID]
+        // Optionally include OpenTrace protocol as scan criteria
+        if BLESensorConfiguration.interopOpenTraceEnabled {
+            scanForServices.append(BLESensorConfiguration.interopOpenTraceServiceUUID)
```

Comment:
Does this throw if enabled is true but UUID still null?

Reply:
Well spotted. This was a potential issue in previous versions of the code.
- All interop configuration values are not optionals. Default values are provided.